net/dns/resolver/tsdns.go

```diff
@@ -725,6 +725,18 @@ func (r *Resolver) resolveLocalReverse(name dnsname.FQDN) (dnsname.FQDN, dns.RCo
 		return "", dns.RCodeRefused
 	}
 
+	// If the requested IP is part of the IPv6 4-to-6 range, it might
+	// correspond to an IPv4 address (assuming IPv4 is enabled).
+	if ip.Is6() && tsaddr.Tailscale4To6Range().Contains(ip) {
+		fqdn, code := r.lookupLocalReverse(tsaddr.Tailscale6to4(ip), name)
+		if code == dns.RCodeSuccess {
+			return fqdn, code
+		}
+	}
+	return r.lookupLocalReverse(ip, name)
```

Comment:
bit of a bummer to acquire that mutex twice in a row, though

Reply:
Yeah, I was originally thinking we could eat the overhead given how rare this path would be. Regardless, ive moved the lock into the calling function.

I tried various ways to structure this better (closure instead of a separate function, a slice of both IPs to lookup and looping through that) and they all seemed like worse vibes.